src/screens/Analytics/GlobalSearch/GlobalSearchBar.res

```diff
@@ -0,0 +1,386 @@
+              {elementsArray
+              ->Array.mapWithIndex(
+                (item, index) => {
+                  let elementValue = item->JSON.Decode.string->Option.getOr("")
+                  <RenderIf condition={elementValue->isNonEmptyString} key={index->Int.toString}>
+                    <RenderedComponent ele=elementValue searchText />
+                    <RenderIf condition={index >= 0 && index < elementsArray->Array.length - 1}>
+                      <span className="mx-2 text-lightgray_background opacity-50">
+                        {">"->React.string}
+                      </span>
+                    </RenderIf>
+                  </RenderIf>
+                },
+              )
+              ->React.array}
+            </OptionWrapper>
+          })
+          ->React.array}
+        </FramerMotion.Motion.Div>
+      })
+      ->React.array}
+    </OptionsWrapper>
+  }
+}
+
+@react.component
+let make = () => {
+  open GlobalSearchTypes
+  open GlobalSearchBarUtils
+  open HeadlessUI
+  open LogicUtils
+  open UIUtils
+  let prefix = useUrlPrefix()
+  let setGLobalSearchResults = GlobalSearchBarUtils.globalSeacrchAtom->Recoil.useSetRecoilState
+  let fetchDetails = APIUtils.useUpdateMethod()
+  let (state, setState) = React.useState(_ => Idle)
+  let (showModal, setShowModal) = React.useState(_ => false)
+  let (searchText, setSearchText) = React.useState(_ => "")
+  let (searchResults, setSearchResults) = React.useState(_ => [])
+  let merchentDetails = HSwitchUtils.useMerchantDetailsValue()
+  let isReconEnabled = merchentDetails.recon_status === Active
+  let hswitchTabs = SidebarValues.useGetSidebarValues(~isReconEnabled)
+  let searchText = searchText->String.trim
+  let loader = LottieFiles.useLottieJson("loader-circle.json")
+  // TODO: need to add feature flag here
+  let isShowRemoteResults =
+    HSLocalStorage.getFromUserDetails("user_role")->String.includes("internal_")
+
+  let redirectOnSelect = element => {
+    let redirectLink = element.redirect_link->JSON.Decode.string->Option.getOr("/search")
+    if redirectLink->isNonEmptyString {
+      setShowModal(_ => false)
+      redirectLink->RescriptReactRouter.push
+    }
+  }
+
+  let getSearchResults = async results => {
+    try {
+      let url = APIUtils.getURL(~entityName=GLOBAL_SEARCH, ~methodType=Post, ())
+      let body = [("query", searchText->JSON.Encode.string)]->LogicUtils.getJsonFromArrayOfJson
+      let response = await fetchDetails(url, body, Post, ())
+
+      let local_results = []
+      results->Array.forEach((item: resultType) => {
+        switch item.section {
+        | Local => local_results->Array.pushMany(item.results)
+        | _ => ()
+        }
+      })
+
+      let remote_results = response->parseResponse
+
+      setGLobalSearchResults(._ => {
+        local_results,
+        remote_results,
+        searchText,
+      })
+
+      let values = response->getRemoteResults
```

Comment:
what is difference between parseResponse and getRemoteResults

Reply:
return types are diff